lib/linguist/samples.rb

```diff
@@ -50,16 +50,13 @@ def self.each(&block)
             end
           else
             path = File.join(dirname, filename)
-
-            if File.extname(filename) == ""
-              raise "#{path} is missing an extension, maybe it belongs in filenames/ subdir"
```

Comment:
Nope, it now allows files without an extension to exist _outside_ of `filenames`.